model/modelprocessor/errormessage_test.go

```diff
@@ -27,39 +27,70 @@ import (
 )
 
-func TestSetErrorMessage(t *testing.T) {
+func TestSetEventMessage(t *testing.T) {
 	tests := []struct {
-		input   *modelpb.Error
-		message string
+		desc                     string
+		input                    *modelpb.APMEvent
+		expectedMessage          string
+		expectedLogMessage       string
+		expectedExceptionMessage string
 	}{{
-		input:   &modelpb.Error{},
-		message: "",
-	}, {
-		input:   &modelpb.Error{Log: &modelpb.ErrorLog{Message: "log_message"}},
-		message: "log_message",
+		desc: "keep event message when no error",
+		input: &modelpb.APMEvent{
+			Error: &modelpb.Error{
+				Log:       &modelpb.ErrorLog{},
+				Exception: &modelpb.Exception{},
+			},
+			Message: "message",
+		},
+		expectedMessage:          "message",
+		expectedLogMessage:       "",
+		expectedExceptionMessage: "",
 	}, {
-		input:   &modelpb.Error{Exception: &modelpb.Exception{Message: "exception_message"}},
-		message: "exception_message",
+		desc: "keep event message when error",
+		input: &modelpb.APMEvent{
+			Error: &modelpb.Error{
+				Log:       &modelpb.ErrorLog{Message: "log_message"},
+				Exception: &modelpb.Exception{Message: "exception_message"},
+			},
+			Message: "message",
+		},
+		expectedMessage:          "message",
+		expectedLogMessage:       "log_message",
+		expectedExceptionMessage: "exception_message",
 	}, {
-		input: &modelpb.Error{
-			Log:       &modelpb.ErrorLog{},
-			Exception: &modelpb.Exception{Message: "exception_message"},
+		desc: "propagate log error if event message empty",
+		input: &modelpb.APMEvent{
+			Error: &modelpb.Error{
+				Log:       &modelpb.ErrorLog{Message: "log_message"},
+				Exception: &modelpb.Exception{},
+			},
+			Message: "",
 		},
-		message: "exception_message",
+		expectedMessage:          "log_message",
+		expectedLogMessage:       "log_message",
+		expectedExceptionMessage: "",
 	}, {
-		input: &modelpb.Error{
-			Log:       &modelpb.ErrorLog{Message: "log_message"},
-			Exception: &modelpb.Exception{Message: "exception_message"},
+		desc: "propagate exception error if event message is empty",
+		input: &modelpb.APMEvent{
+			Error: &modelpb.Error{
+				Log:       &modelpb.ErrorLog{},
+				Exception: &modelpb.Exception{Message: "exception_message"},
+			},
+			Message: "",
 		},
-		message: "log_message",
+		expectedMessage:          "exception_message",
+		expectedLogMessage:       "",
+		expectedExceptionMessage: "exception_message",
 	}}
 
 	for _, test := range tests {
-		batch := modelpb.Batch{{Error: test.input}}
+		batch := modelpb.Batch{test.input}
 		processor := modelprocessor.SetErrorMessage{}
 		err := processor.ProcessBatch(context.Background(), &batch)
 		assert.NoError(t, err)
-		assert.Equal(t, test.message, batch[0].Message)
+		assert.Equal(t, test.expectedMessage, batch[0].Message)
+		assert.Equal(t, test.expectedLogMessage, batch[0].Error.Log.Message)
+		assert.Equal(t, test.expectedExceptionMessage, batch[0].Error.Exception.Message)
```

Comment:
These assertions seem superfluous, since the processor doesn't modify the log/exception message.